ios/Mattermost/MattermostManaged.m

```diff
@@ -59,6 +60,15 @@ - (NSDictionary *)constantsToExport {
             });
 }
 
+RCT_EXPORT_METHOD(isHwKeyboardActive:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  if (@available(iOS 14.0, *)) {
+    
+    resolve(@(GCKeyboard.coalescedKeyboard != nil));
+  } else {
+    resolve(@(NO));
```

Comment:
How does resolving with `@(NO)` compare to `[NSNull null]` or `@{}`?

Reply:
I could return `[NSNull null]` to indicate that the keyboard check isn't available. Or would you suggest an explicit error message instead? I didn't bother with it since we're not separately handling this in the UI 😅

Reply:
I don't have any iOS native code experience so I can't offer any suggestions. I just noticed the other `resolve` params used in the Mattermost code so I'm curious how they compare.

Reply:
Ah okay, I'm assuming you're referring to the dictionaries with boolean properties, like in `isRunningInSplitView`. I can update it to look more like this:
```objective-c
@{
  @"isHwKeyboardAvailable": @(GCKeyboard.coalescedKeyboard != nil), // or @(NO) in the else block
}
```
It feels a bit redundant given the function signature. But I can update it to a dictionary for the sake of consistency.